src/ui/public/courier/saved_object/saved_object.js

```diff
@@ -231,59 +240,48 @@ export default function SavedObjectFactory(es, kbnIndex, Promise, Private, Notif
     };
 
     /**
-     * Save this object
+     * Returns true if the object's original title has been changed. New objects return false.
+     * @return {boolean}
+     */
+    self.isTitleChanged = function () {
+      return self._source && self._source.title !== self.title;
+    };
+
+    /**
+     * Saves this object.
      *
      * @return {Promise}
      * @resolved {String} - The id of the doc
      */
     self.save = function () {
+      // Read https://github.com/elastic/kibana/issues/9056 and
+      // https://github.com/elastic/kibana/issues/9012 for some background into why this copyOnSave variable
+      // exists.
+      // The goal is to move towards a better rename flow, but since our users have been conditioned
+      // to expect a 'save as' flow during a rename, we are keeping the logic the same until a better
+      // UI/UX can be worked out.
+      if (this.copyOnSave) {
+        self.id = null;
+      }
 
-      let body = self.serialize();
-
-      // Slugify the object id
-      self.id = slugifyId(self.id);
-
-      // ensure that the docSource has the current self.id
+      // Create a unique id for this object if it doesn't have one already.
+      this.id = this.id || uuid.v1();
+      // ensure that the docSource has the current id
       docSource.id(self.id);
 
-      // index the document
-      return self.saveSource(body);
-    };
-
-    self.saveSource = function (source) {
-      let finish = function (id) {
-        self.id = id;
-        return es.indices.refresh({
-          index: kbnIndex
-        })
-        .then(function () {
-          return self.id;
-        });
-      };
+      let source = self.serialize();
 
       return docSource.doCreate(source)
```

Comment:
If we are going to automatically call index if we get a 409 then we can just do an index request to begin with. The only difference between the two is that create will fail if the document already exists.